gusto/state.py

```diff
@@ -267,16 +267,22 @@ def setup_diagnostics(self):
 
     def setup_dump(self, pickup=False):
 
-        # setup dump files
         # check for existence of directory so as not to overwrite
         # output files
+        # setup dump files
+        # setup checkpoint files
         self.dumpdir = path.join("results", self.output.dirname)
         outfile = path.join(self.dumpdir, "field_output.pvd")
         if self.mesh.comm.rank == 0 and "pytest" not in self.output.dirname \
            and path.exists(self.dumpdir) and not pickup:
             raise IOError("results directory '%s' already exists" % self.dumpdir)
         self.dumpcount = itertools.count()
         self.dumpfile = File(outfile, project_output=self.output.project_fields, comm=self.mesh.comm)
+        self.chkpts = ["chkpt", "chkptbk"]
+        for name in self.chkpts:
+            chkpt_path = path.join(self.dumpdir, name)
+            chkpt = DumbCheckpoint(chkpt_path, mode=FILE_CREATE)
+            self.chkpts[self.chkpts.index(name)] = chkpt
```

Comment:
FWIW, I would write this loop as:
```python
self.chkpts = []
for name in ["chkpt", "chkptbk"]:
    fullpath = path.join(self.dumpdir, name)
    self.chkpts.append(DumbCheckpoint(fullpath, mode=FILE_CREATE)
```

Reply:
I wrote this so that one can guess right away what's supposed to be in the chkpts list (compared to just having an empty list), but then again looking at your loop I guess it's still completely clear what chkpts is supposed to contain, so I'll replace my loop by yours.